hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -194,6 +194,18 @@ private URL changePrefixFromBlobtoDfs(URL url) throws InvalidUriException {
     return url;
   }
 
+  private URL changePrefixFromDfsToBlob(URL url) throws InvalidUriException {
+    if (url.toString().contains(WASB_DNS_PREFIX)) {
```

Comment:
Can be changed to this :-  if (!url.toString().contains(WASB_DNS_PREFIX)) {
        try {
            url = new URL(url.toString().replace(ABFS_DNS_PREFIX, WASB_DNS_PREFIX));
        } catch (MalformedURLException ex) {
            throw new InvalidUriException(url.toString());
        }
    }
    return url;
}

Reply:
This is same to what is currently there in pr code-change.
Why?:
1. url can either have ABFS_DNS_PREFIX or WASB_DNS_PREFIX. Both are going to be mutually exclusive.
2. For conversion to happen, it expects the string have ABFS_DNS_PREFIX. If is not there, conversion will not happen.
   - There is no need as such to have if condition. But added as if a url already has WASB_DNS_PREFIX, we will be just creating a new redundant URL object in case the if-condition is not there.